organizations/SECRETS.md

```diff
@@ -0,0 +1,50 @@
+# Secret Management
+
+The github foundation terraform modules provide several ways to manage secrets in your github organizations and repositories.
+1. Through the organization module
+2. Through the organization_secrets module
+3. Through the repository_set module
+
+The following sections will cover each of these modules and how to effectively use them.
+
+## Usage
+
+### Encryption
+
+Before using any of the modules to create your github secrets it is important to encrypt them first. Github has documentation outlining how to encrypt secrets for their rest API that can be found here: https://docs.github.com/en/rest/guides/encrypting-secrets-for-the-rest-api
+
+### Organization and organization secrets module
+
+The organiztion and organization secrets modules have the following optional fields for organization secret management:
+- `actions_secrets`: for creating and managing organization secrets accessible to github actions
+- `codespaces_secrets`: for creating and managing organization secrets accessible to github codespaces
+- `dependabot_secrets`: for creating and managing organization secrets accessible to dependabot
+
+Each field takes a map where the key is the name of the secret and the value is an object with the following structure:
+```hcl
+{
+    encrypted_value       = string
+    visibility            = string
+    selected_repositories = optional(list(string))
+}
+```
+
+`encrypted_value` is the encrypted value of the secret you wish to store. See [encryption](#encryption) for instructions on how to encrypt your secret.
+
+`visibility` configures what repositories can access your secret. It must be one of `all`, `private`, or `selected`.
+
+`selected_repositories` is an optional list of repository id strings for which repositories should have access to the secret. Only set this field if visibility has been set to `selected`. It defaults to an empty list if not set so the secret won't be accessible to any repositories if it's not set.
+
+Either module will work well for organization secret management, but in the event that you want to do something like make the terragrunt configuration managing secrets dependant on the terragrunt configuration you are using to create repositories. But don't want the rest of organization management to wait on repository creation before it can execute. Then you might want to consider creating a `secrets` folder and configuration under the `projects/repositories/<ORG-NAME>` path that uses the organization secrets module to create these secrets. Keep in mind that while spreading organization secret definitions out like this can be helpful to keep context of what the secret is used for and where, it also might lead to naming conflicts which would result in one configuration overwriting the secret created by another configuration.
```

Comment:
Should we always suggest the secrets folder?
It makes sense for a lot of reasons. Changes inside it would always be a red flag etc.

Reply:
I think suggesting it for secrets with visibility type `selected` makes a lot of sense. Visibility type `all` doesn't really make that much sense, and I'm not sure about private. And the only thing I worry is losing track of what secret names have already been taken resulting in modules overwriting each others secrets

Reply:
I guess `private` would be like `all` and it doesn't make much sense to put it at the project level. But yeah we could probably reword this to something along the lines of

"When creating secrets that you only want to grant a select amount of repositories access you should consider creating a `secrets` folder...."

Then just warn them about overwrites

Reply:
Oh I think I understand.

I was considering the idea of a "Global" project that we could standardize on.

Something for shared teams, or org secrets.

Reply:
We could also consider a change in how secrets are done. For example, define them all org secrets at the org level. But for visibility of type `select` we set it initially to `[]` then in the repository level we could make use of the tf resources like this https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_organization_secret_repositories to grant them access to the secrets. Would avoid potential overwrite